src/dos/drive_iso.cpp

```diff
@@ -503,7 +503,11 @@ int isoDrive :: readDirEntry(isoDirEntry *de, Bit8u *data) {
 	if (dotpos!=NULL) {
 		if (strlen(dotpos)>4) dotpos[4]=0;
 		if (dotpos-(char*)de->ident>8) {
-			strcpy(reinterpret_cast<char *>(&de->ident[8]), dotpos);
+			constexpr int pos = 8;
+			const auto maxlen = sizeof(de->ident) - pos;
```

Comment:
`de->ident` is a char-array, so this calculation is ok, but if it changed to `char*`, then suddenly this code would have a very silent error.

How about using our `ARRAY_LEN` macro here? If this code will be refactored in the future, we'll have a compile-time check in here.

Reply:
Excellent suggestion, and thank you for the reminder; I think I can put that to use in the other cleanup PRs too! Thank you @dreamer.

Reply:
When I added ARRAY_LEN, I noticed the `- pos` wasn't being subtracted, which was because ARRAY_LEN ends in a semi-colon.  So snipping that fixed it (and fortunately this didn't affect our existing use-cases):

``` text
include/dma.h:          constexpr auto n = ARRAY_LEN(dma_channels);
include/dma.h:          constexpr auto n = ARRAY_LEN(dma_channels);
```